libs/datamodel/core/src/walkers.rs

```diff
@@ -119,21 +119,20 @@ impl<'a> ModelWalker<'a> {
         let model_idx = self.model_idx;
         let datamodel = self.datamodel;
 
-        self.scalar_fields()
-            // Single-id models
-            .filter(|field| field.is_id())
-            // Compound id models
-            .chain(
-                self.get()
-                    .id_fields
-                    .iter()
-                    .filter_map(move |field_name| walker.find_scalar_field(field_name)),
-            )
-            .map(move |field| ScalarFieldWalker {
-                datamodel,
-                model_idx,
-                field_idx: field.field_idx,
-            })
+        let x = if let Some(pk) = &walker.get().primary_key {
```

Comment:
Are we sure we want to allocate a vector in a method that returns an iterator?

Reply:
Option is an iterator, so we could do something like:
```rust
walker.get().primary_key.iter().flat_map(pk.fields.iter()).map(..).map(..)
```

Reply:
I'm going to delete this file "soon".